src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const object of actions) {
+    const dataToRemove = object.keysToRemove;
+    const extraKeysObject = object.extraData;
+
+    if (object.type === 'clear') {
+      Object.keys(state).forEach(key => delete state[key]);
+    };
+
+    if (object.type === 'removeProperties') {
+      Object.keys(state).filter(key => {
+        if (dataToRemove.includes(key)) {
+          delete state[key];
+        }
+      });
+    };
+
+    if (object.type === 'addProperties') {
+      Object.assign(state, extraKeysObject);
+    };
```

Comment:
![image](https://user-images.githubusercontent.com/98764641/215767635-398a6454-e86d-40ab-a057-8dbe967adedd.png)

Reply:
it is from the checklist